app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -73,6 +86,38 @@ class MoneyActivity : SimpleActivity(), Calculator , TaxCalculator {
         }
     }
 
+    override fun spawnCurrencyModal() {
+        currencyDialog = AlertDialog.Builder(this)
+        val currencyDialogView = layoutInflater.inflate(R.layout.currency_modal, null)
+        currencyDialog!!.setView(currencyDialogView)
+        currencyDialog!!.setCancelable(true)
+        var custom_dialog = currencyDialog!!.create()
+        custom_dialog.show()
+        var convert_from = "CAD"
+        var convert_to = "CAD"
+
+        custom_dialog.findViewById<Spinner>(R.id.convert_from).onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+                convert_from = custom_dialog.findViewById<Spinner>(R.id.convert_from).getSelectedItem().toString()
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>) {}
```

Comment:
If nothing is being done here why override this at all ?

Reply:
The listener must implement the abstract method onNothingSelected() to be a concrete object, even if nothing is being done by the method.

Reply:
indeed! Spinners are different than your ListView @almiche